src/tree/LocalRootTreeItemBase.ts

```diff
@@ -318,13 +336,21 @@ export abstract class LocalRootTreeItemBase<TItem extends ILocalItem, TProperty
     }
 
     private async hasChanged(): Promise<boolean> {
+        let pollingDockerStatus: DockerStatus;
+        let isDockerStatusChanged = false;
+
         try {
             this._itemsFromPolling = await this.getSortedItems();
-        } catch {
+            pollingDockerStatus = 'Running';
+        } catch (error) {
             this._itemsFromPolling = undefined;
+            pollingDockerStatus = await this._dockerStatusProvider.isDockerInstalled() ? 'Installed' : 'NotInstalled';
```

Comment:
This is on the UI refresh path. I am worried about performance. Should I be?

Reply:
dockerStatusProvider uses Lazy, so it uses cached value which gets refreshed every 5 seconds. On my machine when the docker is not installed the command takes 10 ms and when docker installed and not running it takes 500 ms. I don't think this will cause any perf issue with the cached value.

Reply:
To be fair, it's only on the _error_ UI refresh path. If we assume that most people have the extension working most of the time, then this will be hit very little.

Reply:
I think we can mitigate the concern by adjusting how `DockerInstallStatusProvider` works